src/Plugin/GraphQL/DataProducer/Entity/EntityLabel.php

```diff
@@ -36,19 +41,22 @@ class EntityLabel extends DataProducerPluginBase implements DataProducerPluginCa
    * Resolver.
    *
    * @param \Drupal\Core\Entity\EntityInterface $entity
+   * @param bool $access
    * @param \Drupal\Core\Session\AccountInterface|null $accessUser
    * @param \Drupal\graphql\GraphQL\Execution\FieldContext $context
    *
    * @return string|null
    */
-  public function resolve(EntityInterface $entity, ?AccountInterface $accessUser, FieldContext $context) {
-    /** @var \Drupal\Core\Access\AccessResultInterface $accessResult */
-    $accessResult = $entity->access('view label', $accessUser, TRUE);
-    $context->addCacheableDependency($accessResult);
-    if ($accessResult->isAllowed()) {
-      return $entity->label();
+  public function resolve(EntityInterface $entity, bool $access, ?AccountInterface $accessUser, FieldContext $context) {
+    if ($access) {
```

Comment:
first we need to check is_null($check_access) and then set it to TRUE (we want access checking on by default).

Reply:
I added the check now.